File: src/types/planner.ts
export type PlannerTaskType = "artist" | "mix" | "master" | "admin";

export type PlannerTask = {
  id: string;
  title: string;
  due_date: string;
  due_time: string | null;
  task_type: PlannerTaskType;
  project_id: string | null;
  notes: string | null;
  completed: boolean;
  created_at_ms: number;
  updated_at_ms: number;
};

export type PlannerTaskInput = {
  id: string;
  title: string;
  dueDate: string;
  dueTime?: string | null;
  taskType: PlannerTaskType;
  projectId?: string | null;
  notes?: string | null;
  completed?: boolean;
};

export type DailyPlanReminderSettings = {
  enabled: boolean;
  time: string;
};

File: src-tauri/src/planner.rs
use serde::{Deserialize, Serialize};

/// One deliberate piece of studio work scheduled in Recall. Dates stay as
/// `YYYY-MM-DD` strings so a task always belongs to the producer's chosen local
/// calendar day instead of drifting when the computer's timezone changes.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PlannerTask {
    pub id: String,
    pub title: String,
    pub due_date: String,
    pub due_time: Option<String>,
    pub task_type: String,
    pub project_id: Option<String>,
    pub notes: Option<String>,
    pub completed: bool,
    pub created_at_ms: u64,
    pub updated_at_ms: u64,
}
